cmd/agent/subcommands/snmp/command.go

```diff
@@ -108,7 +109,33 @@ func Commands(globalArgs *command.GlobalArgs) []*cobra.Command {
 				}
 			} else {
 				deviceIP = address
-				port = defaultPort
+				//if the customer provides only 1 argument : the ip_address
+				//We check the ip address configuration in the agent runtime and we use it for the snmpwalk
+				snmpConfigList, _ := parse.GetConfigCheckSnmp()
+				instance := parse.GetIPConfig(deviceIP, snmpConfigList)
+				if instance.IPAddress != "" {
+					snmpVersion = instance.Version
+					port = instance.Port
+
+					// v1 & v2c
+					communityString = instance.CommunityString
+
+					// v3
+					user = instance.Username
+					authProt = instance.AuthProtocol
+					authKey = instance.AuthKey
+					privProt = instance.PrivProtocol
+					privKey = instance.PrivKey
+					snmpContext = instance.Context
+
+					// communication
+					retries = instance.Retries
+					timeout = instance.Timeout
+
+				} else {
+					deviceIP = address
+					port = defaultPort
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
No Need to reassign `deviceIP`, already done at line 111
```suggestion
					port = defaultPort
```